include/envoy/server/admin.h

```diff
@@ -11,6 +11,9 @@
 namespace Envoy {
 namespace Server {
 
+#define MAKE_HANDLER(X)                                                                            \
+  [this](const std::string& url, Buffer::Instance& data) -> Http::Code { return X(url, data); }
```

Comment:
I don't remember why I originally made this a macro. I think it would be more typing to make it a template and/or factory static method that creates an std::function. Either way you should rename to MAKE_ADMIN_HANDLER if you are moving out to this scope.